File: model/src/main/java/net/contrapt/auction/model/ContactInfo.java
package net.contrapt.auction.model;

import javax.persistence.Embeddable;

/**
 * Created by msimmons on 10/28/14.
 */
@Embeddable
public class ContactInfo {

    public ContactInfo() {
    }

    public ContactInfo(String email) {
        this.email = email;
    }

    public String email;

    public String address1;

    public String address2;

    public String city;

    public String state;

    public String zip;

    public String phone;
}

File: model/src/main/kotlin/net/contrapt/auction/model/ContactInfo.kt
package net.contrapt.auction.model

import javax.persistence.Embeddable

/**
 * Created by msimmons on 10/28/14.
 */
@Embeddable
class ContactInfo : AbstractModel {

    var email: String? = null

    var address1: String? = null

    var address2: String? = null

    var city: String? = null

    var state: String? = null

    var zip: String? = null

    var phone: String? = null

    constructor() {}

    constructor(email: String) {
        this.email = email
    }

}
